src/prefect/client/schemas/objects.py

```diff
@@ -1556,6 +1556,18 @@ class GlobalConcurrencyLimit(ObjectBaseModel):
             " is used as a rate limit."
         ),
     )
+    avg_slot_occupancy_seconds: float = Field(
+        default=2.0,
+        description="The average amount of time a slot is occupied.",
+    )
+    denied_slots: int = Field(
```

Comment:
I'd probably leave denied slots out, it's kind of an internal metric for calculating how long to tell the client to wait before retrying.

Reply:
So I was wondering about this. If I leave these fields out, I get this failing test:

<details>
<pre><code>
 async def test_global_concurrency_limit_update_with_integer(prefect_client):
        # Test for both `integer` and `float` slot_delay_per_second
        for index, slot_decay_per_second in enumerate([1, 1.2]):
            created_global_concurrency_limit_name = (
                f"global-update-test-{slot_decay_per_second}"
            )
            updated_global_concurrency_limit_name = (
                f"global-create-test-{slot_decay_per_second}-new"
            )
            await prefect_client.create_global_concurrency_limit(
                GlobalConcurrencyLimitCreate(
                    name=created_global_concurrency_limit_name,
                    limit=42,
                    slot_decay_per_second=slot_decay_per_second,
                )
            )
            await prefect_client.update_global_concurrency_limit(
                name=created_global_concurrency_limit_name,
                concurrency_limit=GlobalConcurrencyLimitUpdate(
                    limit=1, name=updated_global_concurrency_limit_name
                ),
            )
>           gcl_limits = await prefect_client.read_global_concurrency_limits()

tests/client/test_prefect_client.py:2324: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/prefect/client/orchestration.py:3078: in read_global_concurrency_limits
    return pydantic.parse_obj_as(List[GlobalConcurrencyLimit], response.json())
venvpydantic2/lib/python3.11/site-packages/pydantic/v1/tools.py:38: in parse_obj_as
    return model_type(__root__=obj).__root__
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

__pydantic_self__ = ParsingModel[List[prefect.client.schemas.objects.GlobalConcurrencyLimit]](), data = {'__root__': [{'active': True, 'active_slots': 0, 'avg_slot_occupancy_seconds': 2.0, 'created': '2024-05-01T20:16:04.716835+00:00', ...}]}, values = {}
fields_set = {'__root__'}
validation_error = ValidationError(model='ParsingModel[List[prefect.client.schemas.objects.GlobalConcurrencyLimit]]', errors=[{'loc': ('_...r.extra'}, {'loc': ('__root__', 0, 'denied_slots'), 'msg': 'extra fields not permitted', 'type': 'value_error.extra'}])

    def __init__(__pydantic_self__, **data: Any) -> None:
        """
        Create a new model by parsing and validating input data from keyword arguments.
    
        Raises ValidationError if the input data cannot be parsed to form a valid model.
        """
        # Uses something other than `self` the first arg to allow "self" as a settable attribute
        values, fields_set, validation_error = validate_model(__pydantic_self__.__class__, data)
        if validation_error:
>           raise validation_error
E           pydantic.v1.error_wrappers.ValidationError: 2 validation errors for ParsingModel[List[prefect.client.schemas.objects.GlobalConcurrencyLimit]]
E           __root__ -> 0 -> avg_slot_occupancy_seconds
E             extra fields not permitted (type=value_error.extra)
E           __root__ -> 0 -> denied_slots
E             extra fields not permitted (type=value_error.extra)

venvpydantic2/lib/python3.11/site-packages/pydantic/v1/main.py:341: ValidationError
</code></pre>
</details>

due to updating the Prefect client `read_global_concurrency_limits` method return type. Do you happen to know if we have a better practice for what we should do here?

Reply:
Also, I didn't know you could nest content like that in a comment, very cool.

Reply:
That makes sense, I've added a `GlobalConcurrencyLimitResponse`!